app/Console/Commands/UploadTrainingTickets.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\TrainingTicket;
+use Config;
+use GuzzleHttp\Client;
+use Illuminate\Console\Command;
+use Illuminate\Http\Response;
+
+class UploadTrainingTickets extends Command {
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'VATUSATrainingTickets:UploadPending';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Upload pending training tickets to VATUSA';
+
+    /**
+     * Execute the console command.
+     */
+    public function handle() {
+        $tickets = TrainingTicket::where('vatusa_upload_status', TrainingTicket::$VATUSA_UPLOAD_STATUS["PENDING"])->get();
+
+        foreach ($tickets as $ticket) {
+            $vatusa_position = $this->vatusaizePosition($ticket);
+
+            $req_params = [
+                'form_params' =>
+                [
+                    'instructor_id' => $ticket->trainer_id,
+                    'session_date' => date_format(date_create($ticket->date), "Y-m-d") . ' ' . $ticket->start_time,
+                    'position' => $vatusa_position,
+                    'duration' => $ticket->duration,
+                    'notes' => $ticket->comments,
+                    'location' => 1
+                ],
+                'http_errors' => false
+            ];
+
+            if (is_null($ticket->position)) {
+                continue;
+            }
+
+            $res = (new Client())->request(
+                'POST',
+                Config::get('vatusa.base').'/v2/user/' . $ticket->controller_id . '/training/record?apikey=' . Config::get('vatusa.api_key'),
+                $req_params
+            );
+
+            if ($res->getStatusCode() == Response::HTTP_OK) {
+                $ticket->vatusa_upload_status = TrainingTicket::$VATUSA_UPLOAD_STATUS["UPLOADED"];
+                $ticket->save();
+            }
+        }
+    }
+
```

Comment:
Insert comment here: 
`// Position type must match regex /^([A-Z]{2,3})(_([A-Z]{1,3}))?_(DEL|GND|TWR|APP|DEP|CTR)$/ to be accepted by VATUSA`
Please add this comment just above line 65 in the new UploadTrainingTickets.php. This information is undocumented and led to a multi-day troubleshooting excursion a couple of years ago. I put the information in the code to remind me of this requirement to prevent frustration in the future.

Reply:
Resolved and squashed into bcbd57b